.github/workflows/cicd-main.yml

```diff
@@ -35,40 +35,37 @@ jobs:
   pre-flight:
     runs-on: ubuntu-latest
     outputs:
-      test_to_run: ${{ steps.main.outputs.test_to_run }}
+      test_to_run: ${{ steps.test_to_run.outputs.main }}
+      all: ${{ steps.all.outputs.main }}
     steps:
       - name: Parse test_to_run
-        id: main
+        id: test_to_run
         run: |
           parsed_string=$(echo ${{ inputs.test_to_run }} | jq -c --raw-input 'split(",")')
-          echo "test_to_run=${parsed_string}" >> "$GITHUB_ENV"
+          echo "main=${parsed_string}" | tee -a "$GITHUB_OUTPUT"
+      - name: Parse all
+        id: all
+        run: |
+          echo "main=${{ contains(fromJSON(steps.test_to_run.outputs.main), 'all') }}" | tee -a "$GITHUB_OUTPUT" 
 
   gpu-test:
     needs: [pre-flight]
     runs-on: self-hosted-azure
-    if: ${{ github.event.label.name == 'Run CICD' }}
+    if: ${{ github.event.label.name == 'Run CICD' || github.event_name == 'workflow_dispatch' }}
     steps:
     - name: Run nvidia-smi test
       run: |
         whoami
         nvidia-smi
 
-  cicd-cluster-clean:
+  cicd-test-container-setup:
     runs-on: self-hosted-azure-builder
-    needs: [pre-flight]
-    if: ${{ github.event.label.name == 'Run CICD' }}
+    if: ${{ github.event.label.name == 'Run CICD' || github.event_name == 'workflow_dispatch' }}
     steps:
     - name: Clean server from old files
       run: |
-        docker container prune --filter "until=24h" --force
-        docker image prune -a --filter "until=24h" --force
-
-
-  cicd-test-container-setup:
```

Comment:
There was a reason to keep cicd-cluster-clean separate from cicd-test-container-setup . So let's please get the old form back? :)

Reply:
Sure let me change that back! Out of curiosity, what's the reason to have it separated?

Reply:
Not sure i remember right now out of the top of my mind. Will need to get back to this a bit later

Reply:
Also, better to focus this PR strictly to the changes for the selective trigger purpose :)

Reply:
sure, i readded it